python/ray/serve/_private/application_state.py

```diff
@@ -498,11 +479,53 @@ def _start_or_reconcile_build_app_task(self) -> Tuple[Tuple, BuildAppStatus, str
 
         return None, BuildAppStatus.IN_PROGRESS, ""
 
-    def _reconcile_target_deployments(self) -> None:
+    def _check_deployment_routes(
```

Comment:
It's non-obvious to me that `check_deployment_routes` modifies the `self._route_prefix` instance variable

I'd prefer to structure it to have this method be pure and return the route_prefix if any, then set the route prefix at the callsite (I think we've discussed this pattern before).

Also, what's the purpose of returning an error string then raising in the caller? Why not just raise the exception here?

Reply:
> I'd prefer to structure it to have this method be pure and return the route_prefix if any, then set the route prefix at the callsite (I think we've discussed this pattern before).

Ah yes we have, let me fix that.

> Also, what's the purpose of returning an error string then raising in the caller? Why not just raise the exception here?

This is because of the way the function is also used in `_reconcile_target_deployments` - but I can change it to raise an error and clean it up a bit.